README.md

```diff
@@ -217,6 +221,33 @@ Tells how much to wait in milliseconds until consider the download to be slow.
 
 Width of the image in px.
 
+### observer
+
+> `function({onEnter, onLeave, children})` | optional
+
+A visibility observer component, like react-waypoint, used for lazy-loading.
+
+### children
+
+> `function({onEnter, onLeave, children}) | React.Element` | optional
+
+Observer, used instead of the `observer` prop. Can either be a render prop or a React element
+
+```js
+<IdealImage
+    placeholder={{lqip}}
+    srcSet={[{src: image, width: 3500}]}
+    alt="doggo"
+    width={3500}
+    height={2095}
+    observer={Waypoint}
+>
+  <Waypoint scrollableAncestor="window" /> // onEnter and onLeave are passed automatically
```

Comment:
I started working on this PR mainly because I needed to provide scrollableAncestor="window". Lazy-loading didn't work at all without this. I'm guessing it's either due to SSR or `overflow: auto` in a parent, both common scenarios.

Reply:
It is not because of SSR (SSR should be properly handled)